### DISCUSSION TITLE
Forcing a CSS definition inclusion even when no pages use that definition

### DISCUSSION THREAD

**Discussion by ericjohncarlson** [tailwindlabs/tailwindcss#2708] [Help] [answered]
Hi everyone,
I think this may be a simple problem, but I can't find the answer.  I have a custom css definition defined in my styles.css file, but I only use this definition for content being pulled in from a backend source.  When I run this locally, this is fine.  When I push to Vercel, since that definition isn't used (I guess?), it doesn't end up being defined on my Vercel site.

If I create a test page somewhere and include a single line of text with that definition in use, then the Vercel site works properly.  This is a bit hacky - **is there a tailwind term or similar that I can use to ensure an unused CSS definition is included in the final computed CSS file**?  Thanks!

**Comment by simonswiss** [comment, marked as answer]
Hey!

You can use [the `whitelist` option](https://tailwindcss.com/docs/controlling-file-size#purge-css-options) in the `purge` section of your Tailwind config to achieve that - add the class name(s) you want to preserve even if not present in any of the templates, and they will be skipped by purge!

```diff
// tailwind.config.js
module.exports = {
  purge: {
    content: ['./src/**/*.html'],

    // These options are passed through directly to PurgeCSS
+   options: {
+     whitelist: ['bg-red-500', 'px-4', 'any-class-you-want-to-keep'],
+   }
  },
  // ...
}
```

**Important note**: Tailwind 2.0, which ships in ~2 weeks, updates to PurgeCSS 3, which will lead to a change in this option: the `whitelist` option will become `safelist`, so when you upgrade, you'll need to adjust stat in your config:

```diff
// tailwind.config.js
module.exports = {
  purge: {
    content: ['./src/**/*.html'],

    // These options are passed through directly to PurgeCSS
   options: {
-     whitelist: ['bg-red-500', 'px-4', 'any-class-you-want-to-keep'],
+     safelist: ['bg-red-500', 'px-4', 'any-class-you-want-to-keep'],
   }
  },
  // ...
}
```


Hope it helps! 👍

**Reply by ericjohncarlson** [reply]
Thank you very much, @simonswiss!
